### PR TITLE
Fix ScikitlearnRegressor fit method

### LINKED ISSUE #1536: Bug in fit method of ScikitlearnRegressor
**Describe the bug**
argmax of label is taken before fitting, which shouldn't be done in the case of regression (labels are single values).

**To Reproduce**
1. Create instance of ScikitlearnRegressor
2. Call fit() method of ScikitlearnRegressor with diabetes dataset

**Expected behavior**
model is fitted

**System information (please complete the following information):**
 - OS: MacOS 11.6.2
 - Python version: 3.8
 - ART version or commit number: 1.9.0

### PR DESCRIPTION
**Pull request by abigailgold**
Signed-off-by: abigailt <abigailt@il.ibm.com>

# Description

Remove calling of argmax on labels from fit method. Adjust test to use regression dataset (diabetes).

Fixes #1536

## Type of change

Please check all relevant options.

- [ ] Improvement (non-breaking)
- [ X] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing

Fixed unit test to use regression dataset.

**Test Configuration**:
- OS: 11.6.2
- Python version: 3.8
- ART version or commit number
- TensorFlow / Keras / PyTorch / MXNet version

# Checklist

- [ X] My code follows the style guidelines of this project
- [ X] I have performed a self-review of my own code
- [ ] I have commented my code
- [ ] I have made corresponding changes to the documentation
- [X ] My changes generate no new warnings
- [ X] I have added tests that prove my fix is effective or that my feature works
- [X ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: tests/estimators/regression/test_scikitlearn.py [hunk 1]
```diff
@@ -40,17 +38,15 @@ def setUpClass(cls):
...
     def test_type(self):
         self.assertIsInstance(self.classifier, type(ScikitlearnRegressor(model=self.sklearn_model)))
         with self.assertRaises(TypeError):
             ScikitlearnDecisionTreeRegressor(model="sklearn_model")
 
     def test_predict(self):
-        y_predicted = self.classifier.predict(self.x_test_iris[0:1])
-        y_expected = np.asarray([2.0])
-        np.testing.assert_array_almost_equal(y_predicted, y_expected, decimal=4)
+        _ = self.classifier.predict(self.x_test_diabetes)
```
Comment (beat-buesser) [inline tests/estimators/regression/test_scikitlearn.py:49]: Would it be possible to have again an assert with expected values?

**Review by beat-buesser — CHANGES_REQUESTED** [review]
Hi @abigailgold Thank you very much for your pull request! I have only one question about the changes in the unit test.

**Review by beat-buesser — APPROVED** [review]
